[WebPubSub] Update the README of WebPubSubClient (#33267)
Azure/azure-sdk-for-net: sdk/webpubsub/Azure.Messaging.WebPubSub.Client/README.md

### Authenticate the client

Client uses a Client Access URL to connect and authenticate with the service. The Uri follow the patten as `wss://<service_name>.webpubsub.azure.com/client/hubs/<hub_name>?access_token=<token>`. The client has some different ways to get Client Access URL. As a quick start, you can copy and paste from Azure Portal, and for production, you usually need a negotiation server to generate the Uri.

#### Use Client Access URL from Azure Portal

As a quick start, you can go to the Portal and copy the **Client Access URL** from **Key** blade.

![get_client_url](https://learn.microsoft.com/azure/azure-web-pubsub/media/howto-websocket-connect/generate-client-url.png)

As shown in the diagram, the client will be granted the permission of sending message to the specific group and joining the specific group. Learn more about client permission, see [permissions](https://learn.microsoft.com/azure/azure-web-pubsub/reference-json-reliable-webpubsub-subprotocol#permissions)

```C# Snippet:WebPubSubClient_Construct
var client = new WebPubSubClient(new Uri("<client-access-uri>"));
```

#### Use negotiation server to generate Client Access URL

> In production, client usually fetch Client Access URL from a negotiation server. The server holds the connection string and generates Client Access URL through `WebPubSubServiceClient`. As an sample, the code snippet below just demostrate the how to generate the Client Access URL inside single process.

Reviewer: Shall we add server sdk guide links to let user further discover details? I'm not sure if show `WebPubSubServiceClient` and `WebPubSubClient` together will make it confusing.
Or maybe add a table to introduce the difference clearly?

}
```

> Features to differentiate `WebPubSubClient` and `WebPubSubServiceClient`.

Reviewer 2: this is a little bit confusing, why comparing these 2, how about a graph showing the position of these 2 sdks as what we do in ServiceClient readme

Author: Because we mentioned these two clients in the code sample above. There's a graph contains two clients

### Recovery

> If using reliable protocols, a new WebSocket tries to establish using the connection ID of the lost connection. If the new WebSocket connection is successfully connected, the connection is recovered. And all group contexts will be recovered, and unreceived messages will be resent. If the service returns WebSocket error code `1008` or the recovery attemption lasts more than 30 seconds, the recovery fails.

Reviewer 2: "unreceived messages will be resent" => give it a clearer statement? who unreceived who resend?

### Reconnect

> Reconnection happens when client connection drops and fails to recover. Reconnection just like a new connection which has a new connection ID. After reconnection, the group context or unreceived messages are lost. Client connection needs to rejoin groups. By default, client library rejoin group after reconnection.

Reviewer 2: ```suggestion
Reconnection happens when the client connection drops and fails to recover. Reconnection starts a new connection and the new connection has a new connection ID. After reconnection, the group context or unreceived messages are lost. Client connection needs to rejoin groups. By default, the client SDK rejoins the groups for you after reconnection.
```
